.github/workflows/deploy.yml

```
@@ -1,14 +1,33 @@
-name: Build
+name: Deploy
 on:
   push:
-  pull_request:
+    branches: [master]
+  workflow_dispatch:
+    inputs:
+      ref:
+        description: 'Branch, tag or commit SHA1 to build'
+        required: true
+        type: string
+        default: master
+
+## cancel ongoing runs if the workflow is triggered
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
 jobs:
-  build:
-    name: Build
+  deploy:
+    name: Build and Deploy
     runs-on: ubuntu-latest
+    if: github.repository_owner == 'crystal-lang'
+    env:
+      COMPOSE_FILE: "docker-compose.yml:docker-compose.ci.yml"
+      GIT_REV: "${{ (github.event_name == 'workflow_dispatch' && inputs.ref) || github.ref }}"
```

Am I getting this right?
```suggestion
      GIT_REF: "${{ (github.event_name == 'workflow_dispatch' && inputs.ref) || github.ref }}"
```